observations/obs_converters/BATS/bats_climatology.py

```diff
@@ -0,0 +1,156 @@
+import os
+import numpy as np
+import netCDF4 as nc
+import matplotlib.pyplot as plt
+
+#################################################################
+####################### SCRIPT PARAMETERS #######################
+#################################################################
+
+# input and output datafiles
+input_file_path = 'data/bats_bottle.txt'
+output_txt_path = 'data/bats_climatology.txt'
+output_nc_path  = 'data/bats_climatology.nc'
+
+first_data_line = 61        # file is read starting at this line (so that header information is ignored)
+month_columns   = [18, 19]  # first and last columns in the data file containing month values
+depth_columns   = [58, 63]  # first and last columns in the data file containing depth values
+num_obs_types   = 6         # number of variables being read from the datafile
+max_num_data    = 16660     # upper bound on the number of data points to be processed
+
+# first and last columns in the data file containing values for each variable
+obs_val_columns = [[102, 107],  # O2
+                   [126, 131],  # CO2
+                   [134, 139],  # Alk
+                   [141, 147],  # NO31
+                   [158, 164],  # PO41
+                   [166, 172]]  # Si1
+
+# layer interface depths for data in prog_z.nc output file from MARBL, run "ncdump -v "z_i" prog_z.nc" to generate.
+marbl_depths = [0, 5, 15, 25, 40, 62.5, 87.5, 112.5, 137.5, 175, 225, 275, 350, 450, 
+                550, 650, 750, 850, 950, 1050, 1150, 1250, 1350, 1450, 1625, 1875, 2250, 
+                2750, 3250, 3750, 4250, 4750, 5250, 5750, 6250]
+
+# set to 'True' in order to produce a histogram of BATS observation vs depth for each month
+plot_sample_hist = False
+
+# human-readable names of the variables being read from the file
+obs_names  = ["O2", "CO2", "Alk", "NO31", "PO41", "Si1"]
+
+#################################################################
+####################### MAIN PROGRAM ############################
+#################################################################
+
+clim_means    = np.zeros((num_obs_types, 12, len(marbl_depths)))
+sq_means      = np.zeros((num_obs_types, 12, len(marbl_depths)))
+sample_counts = np.zeros((num_obs_types, 12, len(marbl_depths)), dtype=int)
+
+bats_data   = open(input_file_path, "r")
+line_number = 0
+
+for line in bats_data.readlines():
```

Comment:
Not sure how much python reviewing you want on this,

with open is recommended rather than open and close.
You can slice with readlines to skip the first nlines. 

```
with open(input_file_path, 'r') as bats_data:
    for line in bats_data.readlines()[first_data_line:]
```

Reply:
I like with open more. The code will be modified accordingly